xbmc/games/controllers/types/ControllerHub.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright (C) 2017-2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ControllerHub.h"
+
+#include "ControllerTree.h"
+#include "games/controllers/Controller.h"
+
+#include <utility>
+
+using namespace KODI;
+using namespace GAME;
+
+CControllerHub::~CControllerHub() = default;
+
+CControllerHub& CControllerHub::operator=(const CControllerHub& rhs)
+{
+  if (this != &rhs)
+  {
+    m_ports = rhs.m_ports;
+  }
+
+  return *this;
+}
+
+void CControllerHub::Clear()
+{
+  m_ports.clear();
+}
+
+void CControllerHub::SetPorts(PortVec ports)
+{
+  m_ports = std::move(ports);
+}
+
+bool CControllerHub::IsControllerAccepted(const std::string& controllerId) const
+{
+  bool bAccepted = false;
+
+  for (const CPortNode& port : m_ports)
```

Comment:
This could be solved with an `std::any_of()` call instead of a loop and the `bAccepted` variable. Or just early `return true;` in the loop and `return false` at the end to avoid `bAccepted`. Personally I prefer `std::any_of()`.

Reply:
fixed, and below